core/src/main/java/com/linecorp/armeria/client/AbstractHttpRequestHandler.java

```diff
@@ -147,7 +149,7 @@ public final void operationComplete(ChannelFuture future) throws Exception {
     final boolean tryInitialize() {
         final HttpSession session = HttpSession.get(ch);
         id = session.incrementAndGetNumRequestsSent();
-        if (id >= MAX_NUM_REQUESTS_SENT || !session.canSendRequest()) {
+        if (id >= MAX_NUM_REQUESTS_SENT || !session.canNextRequest()) {
```

Comment:
Let me think again about this change and get back to you

Reply:
I think we don't really need this method. Please the updated test class in the commit I pushed to this PR: 5c2dc573900240946ba3b3ba9619fe68ae8be88c

Reply:
Thanks for your support. I knew what I was misunderstanding.